From 5bc6216346e97eb4bab4a64bfc43c825c67ef9cf Mon Sep 17 00:00:00 2001
From: Sivaprabu Ganesan <sivaprabug@ami.com>
Date: Mon, 12 Jun 2023 14:44:12 +0530
Subject: [PATCH] WEB UI support for NVME Information

Added the WEB UI Page to show the NVMe drive information

Signed-off-by: Sivaprabu Ganesan <sivaprabug@ami.com>
---
 .../AppNavigation/AppNavigationMixin.js       |  8 +++++
 src/env/components/AppNavigation/intel.js     |  8 +++++
 src/env/router/intel.js                       |  9 ++++++
 src/locales/en-US.json                        | 31 +++++++++++++++++--
 src/router/routes.js                          |  9 ++++++
 src/store/index.js                            |  2 ++
 6 files changed, 65 insertions(+), 2 deletions(-)

diff --git a/src/components/AppNavigation/AppNavigationMixin.js b/src/components/AppNavigation/AppNavigationMixin.js
index 42c684a..bae545f 100644
--- a/src/components/AppNavigation/AppNavigationMixin.js
+++ b/src/components/AppNavigation/AppNavigationMixin.js
@@ -7,6 +7,7 @@ import IconSecurity from '@carbon/icons-vue/es/security/16';
 import IconChevronUp from '@carbon/icons-vue/es/chevron--up/16';
 import IconDataBase from '@carbon/icons-vue/es/data--base--alt/16';
 import IconNic from '@carbon/icons-vue/es/network--2/16';
+import IconObjectStorage from '@carbon/icons-vue/es/object-storage/16';
 
 const AppNavigationMixin = {
   components: {
@@ -19,6 +20,7 @@ const AppNavigationMixin = {
     iconExpand: IconChevronUp,
     iconResourceManagement: IconDataBase,
     iconNic: IconNic,
+    iconObjectStorage: IconObjectStorage,
   },
   data() {
     return {
@@ -209,6 +211,12 @@ const AppNavigationMixin = {
           icon: 'iconNic',
           route: '/nic',
         },
+        {
+          id: 'nvmeInformation',
+          label: this.$t('appNavigation.nvmeInformation'),
+          route: '/nvme-information',
+          icon: 'iconObjectStorage',
+        },
       ],
     };
   },
diff --git a/src/env/components/AppNavigation/intel.js b/src/env/components/AppNavigation/intel.js
index 3532382..1d67f6b 100644
--- a/src/env/components/AppNavigation/intel.js
+++ b/src/env/components/AppNavigation/intel.js
@@ -7,6 +7,7 @@ import IconSecurity from '@carbon/icons-vue/es/security/16';
 import IconChevronUp from '@carbon/icons-vue/es/chevron--up/16';
 import IconDataBase from '@carbon/icons-vue/es/data--base--alt/16';
 import IconNic from '@carbon/icons-vue/es/network--2/16';
+import IconObjectStorage from '@carbon/icons-vue/es/object-storage/16';
 
 const AppNavigationMixin = {
   components: {
@@ -19,6 +20,7 @@ const AppNavigationMixin = {
     iconExpand: IconChevronUp,
     iconResourceManagement: IconDataBase,
     iconNic: IconNic,
+    iconObjectStorage: IconObjectStorage,
   },
   data() {
     return {
@@ -204,6 +206,12 @@ const AppNavigationMixin = {
           icon: 'iconNic',
           route: '/nic',
         },
+        {
+          id: 'nvmeInformation',
+          label: this.$t('appNavigation.nvmeInformation'),
+          route: '/nvme-information',
+          icon: 'iconObjectStorage',
+        },
       ],
     };
   },
diff --git a/src/env/router/intel.js b/src/env/router/intel.js
index e871f6b..f98fd1e 100644
--- a/src/env/router/intel.js
+++ b/src/env/router/intel.js
@@ -34,6 +34,7 @@ import Power from '@/views/ResourceManagement/Power';
 import nicInformation from '@/views/Nic';
 import i18n from '@/i18n';
 import Dumps from '@/views/Logs/Dumps';
+import NvmeInformation from '@/views/NvmeInformation';
 
 const routes = [
   {
@@ -307,6 +308,14 @@ const routes = [
           title: i18n.t('appPageTitle.nicInformation'),
         },
       },
+      {
+        path: '/nvme-information',
+        name: 'nvme-information',
+        component: NvmeInformation,
+        meta: {
+          title: i18n.t('appPageTitle.nvmeInformation'),
+        },
+      },
       {
         path: '*',
         name: 'page-not-found',
diff --git a/src/locales/en-US.json b/src/locales/en-US.json
index 4c06506..46b72db 100644
--- a/src/locales/en-US.json
+++ b/src/locales/en-US.json
@@ -137,7 +137,8 @@
     "pef": "Platform Event Filter",
     "eventFilter": "Event Filter",
     "alertDestination": "Alert Destination",
-    "nicInformation": "NIC Information"
+    "nicInformation": "NIC Information",
+    "nvmeInformation": "@:appPageTitle.nvmeInformation"
   },
   "appPageTitle": {
     "changePassword": "Change password",
@@ -172,7 +173,8 @@
     "smtpSettings": "SMTP Settings",
     "eventFilter": "Event Filter",
     "alertDestination": "Alert Destination",
-    "nicInformation": "NIC Information"
+    "nicInformation": "NIC Information",
+    "nvmeInformation": "NVMe Information"
   },
   "pageChangePassword": {
     "changePassword": "Change password",
@@ -1146,6 +1148,31 @@
       "serverClosedWithErrors": "Server closed with errors"
     }
   },
+  "pageNvmeInformation": {
+    "nvmeSolidStateDrive": "NVMe Solid State Drive State",
+    "nvmeInformation":"NVMe Information",
+    "model": "Model",
+    "serialNumber": "Serial number",
+    "pcie0LinkSpeed": "PCIe 0 Link Speed",
+    "pcie1LinkSpeed": "PCIe 1 Link Speed",
+    "pcie0LinkWidth": "PCIe 0 Link Width",
+    "pcie1LinkWidth": "PCIe 1 Link Width",
+    "nvmePowered": "NVMe Powered",
+    "nvmeFunctional": "NVMe Functional",
+    "nvmeResetRequired": "NVMe Reset Required",
+    "port0PcieLinkActive": "Port 0 PCIe Link Active",
+    "port1PcieLinkActive": "Port 1 PCIe Link Active",
+    "deviceClass": "Device Class",
+    "deviceSubClass": "Device Sub-Class",
+    "deviceProgrammingIntfc": "Device Programming Intfc",
+    "driveLifeConsumed": "Drive Life Consumed",
+    "manufacturer": "Manufacturer",
+    "deviceLocation": "Device Location",
+    "firmwareVersion": "firmware Version",
+    "toast": {
+      "errorShowingNVMeInfo": "Error in displaying NVMe Information"
+    }
+  },
   "countries": {
     "AF": "Afghanistan",
     "AL": "Albania",
diff --git a/src/router/routes.js b/src/router/routes.js
index f779e93..3b4a30e 100644
--- a/src/router/routes.js
+++ b/src/router/routes.js
@@ -36,6 +36,7 @@ import VirtualMedia from '@/views/Operations/VirtualMedia';
 import Power from '@/views/ResourceManagement/Power';
 import i18n from '@/i18n';
 import nicInformation from '@/views/Nic';
+import NvmeInformation from '@/views/NvmeInformation';
 
 const routes = [
   {
@@ -325,6 +326,14 @@ const routes = [
           title: i18n.t('appPageTitle.nicInformation'),
         },
       },
+      {
+        path: '/nvme-information',
+        name: 'nvme-information',
+        component: NvmeInformation,
+        meta: {
+          title: i18n.t('appPageTitle.nvmeInformation'),
+        },
+      },
       {
         path: '*',
         name: 'page-not-found',
diff --git a/src/store/index.js b/src/store/index.js
index d2a6318..10d9937 100644
--- a/src/store/index.js
+++ b/src/store/index.js
@@ -38,6 +38,7 @@ import VirtualMediaStore from './modules/Operations/VirtualMediaStore';
 import KvmStore from './modules/Operations/KvmStore';
 import AlertDestinationStore from './modules/Pef/alertDestinationStore';
 import NicStore from './modules/Nic/Nic';
+import NvmeInformationStore from './modules/NvmeInformation/NvmeInformationStore';
 
 Vue.use(Vuex);
 
@@ -82,6 +83,7 @@ export default new Vuex.Store({
     keyClear: KeyClearStore,
     FactoryDefault: FactoryDefaultStore,
     nic: NicStore,
+    nvme: NvmeInformationStore,
   },
   plugins: [WebSocketPlugin],
 });
-- 
2.25.1

